sydent/terms/terms.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import yaml
+
+
+logger = logging.getLogger(__name__)
+
+
+class Terms(object):
+    def __init__(self, yamlObj):
+        self._rawTerms = yamlObj
+
+    def getMasterVersion(self):
+        return None if self._rawTerms is None else self._rawTerms['master_version']
+
+    def getForClient(self):
+        policies = {}
+        if self._rawTerms is not None:
+            for docName, doc in self._rawTerms['docs'].items():
+                policies[docName] = {
+                    'version': doc['version'],
+                }
+                for langName, lang in doc['langs'].items():
+                    policies[docName][langName] = lang
+        return { 'policies': policies }
+
+    def getUrlSet(self):
+        urls = set()
+        if self._rawTerms is not None:
+            for docName, doc in self._rawTerms['docs'].items():
+                for langName, lang in doc['langs'].items():
+                    urls.add(lang['url'])
```

Comment:
fwiw this can be written as:

```python
urls = set(
    lang['url']
    for doc in self._rawTerms['docs'].values()
    for lang in doc['langs'].values()
)
```